File: Week_02/Readability.c
/**********************************************************************/
/*                          Week 2 : Readability                      */
/*                                                                    */
/*   Implement a program in C that prompts the user for their name    */
/*                 and then says hello to that user.                  */
/**********************************************************************/

#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int main(){
    string text = get_string("Text: ");
    int sentence = 0, word = 1, n = strlen(text), letter = 0;
    for(int i = 0; i < n; i++){
        if(text[i] == ' '){
            word++;
        }
        else if((text[i] == '.')||(text[i] == '!') || (text[i] == '?')){
            sentence++;
        }
    }
    for(int j = 0; j < n; j++){
        if(isalpha(text[j])){
            letter++;
        }
    }
    float resultat = (0.0588 * (((float)letter/(float)word)*100) - 0.296 * (((float)sentence/(float)word)*100) - 15.8);
    int r = round(resultat);
    if(r <= 0){
        printf("Before Grade 1\n");
    }
    else if(r >= 16){
        printf("Grade 16+\n");
    }
    else{
        printf("Grade %i\n", r);
    }
    return 0;
}

File: Week_06/Readability.py


from cs50 import get_string
import math

text = get_string("text: ")
sentence = 0
word = 1
n = len(text)
letter = 0

for i in range(n):
    if text[i] == ' ':
        word += 1

    elif(text[i] == '.') | (text[i] == '!') | (text[i] == '?'):
        sentence += 1

    elif (text[i].isalpha()):
        letter +=1

r = (0.0588 * (letter/word)*100) - 0.296 * ((sentence/word)*100) - 15.8
r = round(r)

#print("sentence",sentence)
#print("word:",word)
#print("letter:",letter)

if r <= 0:
    print("Before Grade 1")

elif r >= 16:
    print("Grade 16+")

else:
    print("Grade", r)
